test/types/models.test.ts

```diff
@@ -751,3 +751,22 @@ function gh13957() {
   const repository = new RepositoryBase<ITest>(TestModel);
   expectType<Promise<ITest[]>>(repository.insertMany([{ name: 'test' }]));
 }
+
+function gh13897() {
+  interface IDocument {
+    name: string;
+    createdAt: Date;
+    updatedAt: Date;
+  }
+
+  const documentSchema = new Schema<IDocument>({
+    name: { type: String, required: true }
+  },
+  {
+    timestamps: true
+  });
+
+  const Document = model<IDocument>('Document', documentSchema);
+  const doc = new Document({ name: 'foo' });
```

Comment:
if i understand correctly, before this change this would complain that `createdAt` and `updatedAt` are missing?

if yes, with the way it is currently implemented, maybe add a test that tests for error with `new Document<IDocument>({ name: 'foo' });` (or the proper doc type for here)

Reply:
@hasezoey fixed, can you please re-review?